desktop/flipper-plugin/src/ui/DataFormatter.tsx

```diff
@@ -187,3 +198,31 @@ const truncateButtonStyle = {
   color: theme.primaryColor,
   marginLeft: 4,
 };
+
+type HljsStylesType = typeof hljsStyles;
+
+type SyntaxHighlightProps = {
+  children: React.ReactNode;
+};
+
+function SyntaxHighlight({children}: SyntaxHighlightProps) {
+  const {
+    enabledLogsSyntaxHighlighting,
+    logsSyntaxHighlightingStyle = 'atomOneLight',
+  } = window.FlipperRenderHostInstance.serverConfig.settings;
```

Comment:
Instead of using RenderHost directly, we expose [`FlipperLib`](https://github.com/facebook/flipper/blob/main/desktop/flipper-plugin/src/plugin/FlipperLib.tsx) to plugins. 
Could we extend it instead?
```ts
interface FLipperLib {
  logsSyntaxHighlighting?: {
    enabled: boolean,
    style: keyof HljsStylesType
  }
}
```
We could pass settings from render host to flipper lib [here](https://github.com/facebook/flipper/blob/main/desktop/flipper-ui-core/src/utils/flipperLibImplementation/index.tsx#L39)

Reply:
I am also wondering if we should check it in `highlightSyntax` and remove the checks from the component itself?

Reply:
Thank you for your review @aigoncharov. I have the checks inside the component, because there we need to get the setting for the selected highlighting theme, and I thought, since we're going to use the settings there, why not check for both *theme* and *enabled* flag. I'll put it inside the formatter instead and leave the theme setting inside the component; or maybe fetch *theme* setting even inside the `highlightSyntax` formatter and pass it as a prop to the `SyntaxHighlight` component? Maybe that's even better.

Reply:
I'll also try to extend `FlipperLib` as you suggested and update here. Hope I'll manage to address all of your points today to forward this PR.